misc/toml-parser/src/main/java/io/ballerina/toml/semantic/ast/TomlTransformer.java

```diff
@@ -338,24 +340,93 @@ private TomlNodeLocation getPosition(Node node) {
 
     @Override
     public TomlNode transform(StringLiteralNode stringLiteralNode) {
-        String valueString = stringLiteralNode.content().text();
+        boolean multilineString = isMultilineString(stringLiteralNode.startDoubleQuote());
+        Optional<Token> content = stringLiteralNode.content();
+        String valueString;
+        if (content.isEmpty()) {
+            valueString = "";
+        } else {
+            valueString = content.get().text();
+            if (multilineString) {
+                valueString = removeFirstNewline(valueString);
+                valueString = trimBackslashWhitespaces(valueString);
+            }
+        }
         String unescapedJava = StringEscapeUtils.unescapeJava(valueString);
         TomlNodeLocation position = getPosition(stringLiteralNode);
 
         return new TomlStringValueNode(unescapedJava, position);
     }
 
+    private String trimBackslashWhitespaces (String value) {
+        StringBuilder output = new StringBuilder();
+        String[] split = value.split("\\\\\\r?\\n");
+        for (String str : split) {
+            output.append(str.stripLeading());
+        }
+        return output.toString();
+    }
+
+    @Override
+    public TomlNode transform(LiteralStringLiteralNode literalStringLiteralNode) {
+        boolean multilineString = isMultilineString(literalStringLiteralNode.startSingleQuote());
+        Optional<Token> content = literalStringLiteralNode.content();
+        String valueString;
+        if (content.isEmpty()) {
+            valueString = "";
+        } else {
+            valueString = content.get().text();
+            if (multilineString) {
+                valueString = removeFirstNewline(valueString);
+            }
+        }
+        TomlNodeLocation position = getPosition(literalStringLiteralNode);
+
+        return new TomlStringValueNode(valueString, position);
+    }
+
+    private boolean isMultilineString(Token token) {
+        return token.kind().equals(SyntaxKind.TRIPLE_SINGLE_QUOTE_TOKEN) ||
+                token.kind().equals(SyntaxKind.TRIPLE_DOUBLE_QUOTE_TOKEN);
```

Comment:
can use == to check. (it can prevent possible NPE, complie time type safe and faster compared to a method call)